sdcm/nemesis.py

```diff
@@ -4904,6 +4905,121 @@ def are_gate_closed_messages_raised(log_reader):
         assert not gate_closed_appearing, \
             "After re-enabling binary and gossip, 'gate closed' messages continue to appear"
 
+    def disrupt_refusing_connection_from_banned_node(self, use_iptables=False):
+        """Banned node could not connect with rest nodes in cluster
+
+        If node was removed from cluster for any reason, even if removed node
+        become alive and try to communicate with rest node in cluster, all connections
+        from it should be refused by other nodes in cluster
+        1. on target node block any scylladb process
+        1.1 Pause process
+        1.2 Block with iptables port 9100/10000
+        2. Wait and remove target node from cluster
+        3. start scylla on target node
+        4. Create exclusive connection to target node
+        5. Execute cql command on target node and validate that no operation
+        from target node passed to cluster
+        """
+        if not self.target_node.raft.consistent_topology_changes_enabled:
+            raise UnsupportedNemesis("Experimental feature 'consistent-topology-changes' is not enabled")
+
+        def prepare_test_keyspace(node):
+            """ Create new table for verification missing data
+
+            Create new keyspace and table which allow to
+            verify that no operations come from banned node
+            """
+            with self.cluster.cql_connection_patient(node) as session:
+                session.execute(
+                    "CREATE KEYSPACE IF NOT EXISTS banned_keyspace \
+                        WITH replication = {'class': 'SimpleStrategy', 'replication_factor' : 3};")
+                session.execute("CREATE TABLE IF NOT EXISTS banned_keyspace.table1 (id BIGINT PRIMARY KEY, name text);")
+
+        def read_barrier(node):
+            """ issue read barrier on node to check that
+            node topology and schema are updated"""
+            with self.cluster.cql_connection_patient(node) as session:
+                session.execute("DROP TABLE IF EXISTS noexist_ks.noexist_cf;")
+
+        @contextlib.contextmanager
+        def use_network_firwall(target_node: BaseNode):
```

Comment:
I propose to separate `simulate_node_unavailability` context manager to separate file so we don't extend more `nemesis.py` file. Also it will enhance SCT framework with reusable functionality.
My proposal:
```
def simulate_node_unavailability(node: BaseNode, using_iptables: bool = False)
    if using_iptables:
        return use_network_firewall(node)
    else:
        return use_stop_signal(node)
```

We even could go further by adding `wait_for_cluster_ack_node_is_down` argument to this method as it seems to be quite common and further reduce `nemesis.py` complexity.

Reply:
Fixed